### ISSUE TITLE
Not compatible with Bartender or Vanilla menu bar tidy tools on Mac

### ISSUE THREAD

**Issue by cquilboss** [docker/for-mac#3122]
The Docker for Mac app is not compatible with Bartender or Vanilla Apps that are are menu bar tidy tools.

  - [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 98A39DCC-A9BE-41CC-9782-092D7A40A3E4/20180726201738

### Expected behavior
Docker menu bar icon should hide or be controllable by third parties like Bartender or Vanilla.

### Actual behavior
Currently, the Docker Mac App menu bar icon is always visible and cannot be temporarily hidden as any other app. Example with Bartender App: https://prnt.sc/kbgrih

### Information
Issue can be reproduced, this problem is old and remains on macOS 10.13.6 or below.
Issue happens with only few other app as described in the FAQ: https://www.macbartender.com/b3_knowledge/iHeartRadio-app-not-controllable/

### Diagnostic logs
Best is the screenshot to understand the issue. We should not see the Docker icon: https://prnt.sc/kbgrih

### Steps to reproduce the behavior
1. Install either Bartender https://www.macbartender.com or Vanilla https://matthewpalmer.net/vanilla/
2. Try to manage or hide the Docker icon

**Comment by Denis4yk** [comment]
+1

**Comment by tino-codes** [comment]
+1

Even if it only concerns a small circle of users: please fix this. The behavior is really annoying.

**Comment by juliovedovatto** [comment]
+1

I would like to be able to hide Docker icon. The behavior is really annoying.

**Comment by lijunyi0198** [comment]
+1

**Comment by DavidMay371** [comment]
+1

**Comment by timuster** [comment]
+1

**Comment by safronov3d** [comment]
+1

**Comment by steakknife** [comment]
Extremely annoying

**Comment by rubenvde** [comment]
+1

**Comment by CelsoSantos** [comment]
I'm joining the chorus here. I NEED this.

**Comment by lizeyan** [comment]
+1

**Comment by gtaranti** [comment]
+1

**Comment by alfuken** [comment]
+1 🆙

**Comment by fliaping** [comment]
+1

**Comment by ivanmauricio** [comment]
👍

**Comment by AaronGillBraun** [comment]
👍

**Comment by joepenzo** [comment]
+1

**Comment by RalphRouwen2** [comment]
+1

**Comment by joseeduardoperes** [comment]
I have the same problem with Docker 2.1.00 (36874) and Bartender 3.0.64. Is there any fix for that?

**Comment by alfuken** [comment]
Looks like nobody just gives a cares about it.

**Comment by joseeduardoperes** [comment]
I think I found a solution for that!
I quitted Bartender, then I manually moved the Docker icon (⌘+dragging), then I started Bartender again, and the Docker icon appeared. :)

**Comment by alfuken** [comment]
@joseeduardoperes  Nope, still can't control it.
<img width="467" alt="Screenshot 2019-07-31 at 19 38 18" src="https://user-images.githubusercontent.com/75377/62234332-c74d2c00-b3ca-11e9-8c49-0be4b288206a.png">

**Comment by joseeduardoperes** [comment]
@alfuken yes unfortunately the problem persists..., but at least now the Docker icon shows on Bartender, before doing that, I couldn't show the Docker icon when Bartender was open. Waiting for a "full" solution..

**Comment by CelsoSantos** [comment]
> I found a "partial" solution for that, at least for who wants to show the Docker icon using Bartender.
> I quitted Bartender, then I manually moved the Docker icon (⌘+dragging), then I started Bartender again, and the Docker icon appeared. :)

I'm not entirely sure if it was caused by upgrading to Bartender 3.1.1, the Docker 2.1.0 upgrade or because of that process, but somehow, Docker now shows on my menubar even with Bartender running.

**Comment by juliovedovatto** [comment]
I can confirm: latest Bartender update solved the problem. Now Docker is hidden yay 🎉

1 year later ... damn

![screenshot 2019-08-01 às 17 36 34](https://user-images.githubusercontent.com/320507/62325831-f7f09c80-b482-11e9-8006-b289e3cdf83e.png)

**Comment by joseeduardoperes** [comment]
> I can confirm: latest Bartender update solved the issue. Now Docker is hidden yay 🎉
> 
> 1 year later ... damn
> 
> Even with Docker in latest version. I tested closing and opening it and after a fresh OS reboot.
> 
> ![screenshot 2019-08-01 às 17 36 34](https://user-images.githubusercontent.com/320507/62325831-f7f09c80-b482-11e9-8006-b289e3cdf83e.png)

Just to make sure..., which versions exactly are you running of macOS, Docker and Bartender?

**Comment by alfuken** [comment]
I can conform too, updating to Bartender 3.1.1 resolves the issue. Still bugged in Vanilla, tho.

**Comment by jellycsc** [comment]
Docker in Vanilla is working fine.

**Comment by alexgurr** [comment]
Doesn't seem to work for me. Latest bartender 3.1.2 + docker 2.1.0.2

**Comment by KTamas** [comment]
this is happening again with Docker for Mac and Bartender 3.1.6

**Comment by rashad** [comment]
+1 for Docker for Mac 2.2.0.4 and Bartender 3.1.6

**Comment by ajota-admios** [comment]
+1

**Comment by qin-guan** [comment]
+1

**Comment by Edger** [comment]
+1

**Comment by tib-kpl** [comment]
+1

**Comment by mcao2** [comment]
+1

**Comment by zohuchneg** [comment]
+1

**Comment by henrycodebreaker** [comment]
+1

**Comment by mmcalli** [comment]
I just had this problem with Docker 3.0.4 (51218) and Bartender 3.1.25. Resolved it using the following sequence of steps:-

1. Quit Bartender
2. ⌘_dragging dockers menu bar icon to a different position
3. Restart Bartender

On restart, Bartender had control over Docker.

**Comment by ahmadrio** [comment]
> I found a "partial" solution for that, at least for who wants to show the Docker icon using Bartender.
> I quitted Bartender, then I manually moved the Docker icon (⌘+dragging), then I started Bartender again, and the Docker icon appeared. :)

thanks it works
